crates/router/src/connector/authorizedotnet/transformers.rs

```diff
@@ -636,7 +638,7 @@ pub struct AuthorizedotnetErrorResponse {
 
 fn construct_refund_payment_details(masked_number: String) -> PaymentDetails {
     PaymentDetails::CreditCard(CreditCardDetails {
-        card_number: masked_number.into(),
+        card_number: cards::CardNumber::from_str(&masked_number).unwrap(),
         expiration_date: "XXXX".to_string().into(),
         card_code: None,
     })
```

Comment:
@Narayanbhat166 Would this fail if the card number passed to the function is a masked one? How do you suggest we handle this?

Reply:
In case of authorizedotnet, the card number that we receive in the response will be masked already, it will be in the format `1234XX`. Validating this will always fail. You should change the type of card number ( only for `authorizedotnet` ) to not have any validation. 

This case is special for this connector because it expects the first four digits of the card number to be sent in the `refunds` call. In `payments` response it sends a masked card number which we need to send in the refunds call. So we save this data in the `payment_attempt`.